### PR TITLE
Unified conversation list

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Update the UI to be compatible with the changes made to the data model in https://github.com/wireapp/wire-ios-data-model/pull/288.
* There are still a couple of important TODOs as this was more about making the project compile again.

### This PR depends on:
- [x] A new release of `wire-ios-data-model` after https://github.com/wireapp/wire-ios-data-model/pull/288 has been merged.
- [x] A new release of `wire-ios-share-engine` after https://github.com/wireapp/wire-ios-share-engine/pull/33 has been merged.
- [x] A new release of `wire-ios-sync-engine` after https://github.com/wireapp/wire-ios-sync-engine/pull/415 has been merged.
- [x] A new release of `wire-ios-message-strategy` after https://github.com/wireapp/wire-ios-message-strategy/pull/77 has been merged.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -53,7 +53,7 @@ CHECKOUT OPTIONS:
 
 SPEC CHECKSUMS:
   ARCollectionViewMasonryLayout: 97cb468434b546b79a98c7c0908447ff803eb85e
-  GoogleAPIClient: 4dedb6af613eb7d171814c343f8a82ee5411e13b
+  GoogleAPIClient: bde49b31072d3c296d051f58c37b27a66ca45936
```
Comment (daehn) [inline Podfile.lock]: Not so sure about this 🤔

File: Wire-iOS/Sources/UserInterface/UserProfile/ProfileDetailsViewController.m [hunk 2]
```diff
@@ -162,7 +161,7 @@ - (void)createTeamsGuestLabel
     self.teamsGuestLabel.numberOfLines = 0;
     self.teamsGuestLabel.textAlignment = NSTextAlignmentCenter;
     [self.userImageViewContainer addSubview:self.teamsGuestLabel];
-    self.teamsGuestLabel.text = [self teamGuestsTextFor:self.bareUser];
+    // self.teamsGuestLabel.text = [self teamGuestsTextFor:self.bareUser]; TODO: Fix this
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/UserProfile/ProfileDetailsViewController.m]: This is still needed, but maybe the text could simply say `Guest` as there is only one team anyway. Better to check with @Yorgos-V

File: Wire-iOS/Sources/UserInterface/UserProfile/ProfileDetailsViewController.swift [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-//
-
-import Foundation
-import WireDataModel
-
-extension ProfileDetailsViewController {
-    @objc public func teamGuestsText(for user: ZMBareUser) -> String {
-        return user.guestInTeams.flatMap {
-            guard let teamName = $0.name else { return nil }
-            return "profile.details.guest_in_team".localized(args: teamName)
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/UserProfile/ProfileDetailsViewController.swift]: Would be worth removing from `.strings` file as well

File: Wire-iOS/Sources/UserInterface/UserProfile/ProfileDetailsViewController.m (hunk 2 above)
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/UserProfile/ProfileDetailsViewController.m]: Yeah, that's what I thought, will check with @Yorgos-V.

**Comment by daehn** [comment]
@vytis @mikeger also updated the share extension (removed the team picker for now).

**Comment by daehn** [comment]
@vytis @mikeger updated the tests, now all ✅

**Comment by mikeger** [comment]
@daehn please release the hounds, ekhm, no, merge and release the frameworks.

**Comment by daehn** [comment]
@mikeger on it 🐙

**Pull request merged**